### PR TITLE
Update the view when sending an update. Disable loopback

### PR DESCRIPTION
**Pull request by samiwh**
#### Description
Sending an update will update the view. This does not make the more intrusive changes, like separating the writer send api from the ecs one. Also does not deal with the potential `in` copy issue.

### CODE REVIEW

File: workers/unity/Packages/com.improbable.gdk.core/Worker/SerializedMessagesToSend.cs [hunk 1]
```diff
@@ -112,7 +115,7 @@ public void SendAll(Connection connection, CommandMetaData commandMetaData)
             for (int i = 0; i < updates.Count; ++i)
             {
                 ref readonly var update = ref updates[i];
-                connection.SendComponentUpdate(update.EntityId, update.Update);
+                connection.SendComponentUpdate(update.EntityId, update.Update, updateParams);
```
Comment (jamiebrynes7) [inline workers/unity/Packages/com.improbable.gdk.core/Worker/SerializedMessagesToSend.cs]: Isn't this a semantic breaking change?

File: workers/unity/Packages/com.improbable.gdk.core/View/View.cs [hunk 2]
```diff
@@ -136,5 +122,26 @@ public bool IsAuthoritative(EntityId entityId, uint componentId)
...
+
+            var entitiesRemoved = diff.GetEntitiesRemoved();
+            foreach (var entity in entitiesRemoved)
+            {
+                entities.Remove(entity);
+            }
+
+            foreach (var storage in viewStorages)
+            {
+                // Resolve this with an actual diff!
```
Comment (zeroZshadow) [inline workers/unity/Packages/com.improbable.gdk.core/View/View.cs]: Nit: No need for the !

File: workers/unity/Packages/com.improbable.gdk.core/Worker/SerializedMessagesToSend.cs [hunk 3]
```diff
@@ -37,8 +37,11 @@ public class SerializedMessagesToSend
         private readonly List<IComponentSerializer> componentSerializers = new List<IComponentSerializer>();
         private readonly List<ICommandSerializer> commandSerializers = new List<ICommandSerializer>();
 
+        private readonly UpdateParameters updateParams = new UpdateParameters();
+
         public SerializedMessagesToSend()
         {
+            updateParams.Loopback = ComponentUpdateLoopback.None;
```
Comment (zeroZshadow) [inline workers/unity/Packages/com.improbable.gdk.core/Worker/SerializedMessagesToSend.cs]: Can this not be set in the object's constructor?

File: workers/unity/Packages/com.improbable.gdk.core/View/View.cs (hunk 2 above)
Comment (samiwh) [reply workers/unity/Packages/com.improbable.gdk.core/View/View.cs]: I just copied this over.

File: workers/unity/Packages/com.improbable.gdk.core/Worker/SerializedMessagesToSend.cs (hunk 3 above)
Comment (samiwh) [reply workers/unity/Packages/com.improbable.gdk.core/Worker/SerializedMessagesToSend.cs]: It doesn't have a non default constructor.

File: workers/unity/Packages/com.improbable.gdk.core/View/View.cs (hunk 2 above)
Comment (samiwh) [reply workers/unity/Packages/com.improbable.gdk.core/View/View.cs]: @jamiebrynes7 what is this comment for

File: workers/unity/Packages/com.improbable.gdk.core/Worker/SerializedMessagesToSend.cs (hunk 1 above)
Comment (samiwh) [reply workers/unity/Packages/com.improbable.gdk.core/Worker/SerializedMessagesToSend.cs]: Actually that is only half true. We don't reapply the update to the component, but we do add it to the updates received list. So these might need to stay :(. We currently have nothing internal depending on this behaviour but a user might. Basically monobehaviours should have an immediate update or we should keep a list of updates sent as well as received.

File: workers/unity/Packages/com.improbable.gdk.core/View/View.cs (hunk 2 above)
Comment (jamiebrynes7) [reply workers/unity/Packages/com.improbable.gdk.core/View/View.cs]: Leftover comment that I forgot to remove. Can remove

File: workers/unity/Packages/com.improbable.gdk.core/Worker/SerializedMessagesToSend.cs (hunk 1 above)
Comment (samiwh) [reply workers/unity/Packages/com.improbable.gdk.core/Worker/SerializedMessagesToSend.cs]: Hmm will keep the loopback for the moment although we really should get rid of it.

File: workers/unity/Packages/com.improbable.gdk.core/Worker/SerializedMessagesToSend.cs (hunk 3 above)
Comment (zeroZshadow) [reply workers/unity/Packages/com.improbable.gdk.core/Worker/SerializedMessagesToSend.cs]: If its a struct you can just use named initialisation.

Comment (samiwh) [reply workers/unity/Packages/com.improbable.gdk.core/Worker/SerializedMessagesToSend.cs]: It's a class. It has to be set after it's definition.

**Pull request merged**